.travis.yml

```diff
@@ -58,6 +58,7 @@ before_script:
  - if ! [ $(phpenv version-name) = "5.3" ]; then printf "\n" | travis_retry pecl install imagick; fi
  - if [ $(phpenv version-name) = "5.3" ]; then printf "\n" | travis_retry pecl install imagick-3.3.0; fi
  - composer self-update || true
+ - if [ $INSTALL_PHPUNIT_FORK ]; then composer require --no-update --dev sminnee/phpunit-mock-objects:^3.4.7; fi
```

Comment:
The fork is not compatible with PHP 5.4 and PHP 5.5. So instead of requiring all the time, I'll install it only when we're testing 7.4.

Reply:
If this is going to be a requirement for 7.4 compatibility going ahead shouldn't we add this fork to our dev constraints (rather than just get it working for our CI but nobody actually using 7.4 for dev)?

Reply:
If I do that, we've got to remove the PHP 5.4 and PHP 5.5 entry in the test matrix.

When I talked to @sminnee and @chillu about it, they were OK with removing PHP5.4/5.5.

I guess we could try to call `composer remove --dev sminnee/phpunit-mock-objects` on those two matrix entry instead.

Reply:
I personally would prefer we just drop 5.4 and 5.5 and require this module in our dev constraints. People using the latest version of PHP should get a copy of the framework they can develop on by default.

It would make it very difficult for someone running on PHP 5.4 or 5.5 though because it's hard to un-install a dependency with composer. 🤔 

I don't know the best answer to be honest, I'll leave it for more opinion

Reply:
I would note that `require-dev` dependencies have no impact either way on users of the package; it only affects those who are developing the package. So outside of core devs responsible for keeping SS3.7 running on PHP7.4 (which will probably be a handful of SSLtd staff in 6-12 months time), either solution is identical.

Note that this means that for a user wishing to upgrade their site or module to PHP 7.4 _and_ run tests on that side that make use of the PHPUnit mocking system, they will need to manually install sminnee/phpunit-mock-objects. This will be the case even if sminnee/phpunit-mock-objects is added as a dev dependencies (as it is in SS4).

In my view, this is best provided as installation notes and/or an 'upgrading to php 7.4' guide.

As such, either Max's current solution of Guys' recommended changes will work. I support either, but I do think that Guy is overstating the downsides of the current solution because of the above.

Reply:
Ah yep. I often forget that. It's a good point - let's just merge this.